gcc/rust/expand/rust-macro-substitute-ctx.cc

```diff
@@ -199,8 +200,13 @@ std::pair<std::vector<std::unique_ptr<AST::Token>>, size_t>
 SubstituteCtx::substitute_token (size_t token_idx)
 {
   auto &token = macro.at (token_idx);
+
+  if (token_id_is_keyword (token->get_id ()))
+    goto case_keyword;
```

Comment:
do you need this goto? 

I wonder can you turn the case IDENTIFIER into the default case and leave the case LEFT_PAREN as is?